Source/Scene/Cesium3DTilesetAsyncTraversal.js

```diff
@@ -0,0 +1,141 @@
+define([
+        '../Core/Intersect',
+        '../Core/ManagedArray',
+        './Cesium3DTileRefine'
+    ], function(
+        Intersect,
+        ManagedArray,
+        Cesium3DTileRefine) {
+    'use strict';
+
+    /**
+     * Traversal that loads all leaves that intersect the camera frustum.
+     * Used to determine ray-tileset intersections during a pickFromRayMostDetailed call.
+     *
+     * @private
+     */
+    function Cesium3DTilesetAsyncTraversal() {
+    }
+
+    var asyncTraversal = {
+        stack : new ManagedArray(),
+        stackMaximumLength : 0
+    };
+
+    Cesium3DTilesetAsyncTraversal.selectTiles = function(tileset, frameState) {
+        tileset._selectedTiles.length = 0;
```

Comment:
Given that this is reaching into tileset private variables, should those be broken out into a separate class that the tileset still owns but is just properties used for traversal?  Would reduce coupling, but is it worth it?

Reply:
Probably the right thing to do, but I'd want to go all in on this and fix it throughout all the 3D Tiles files. I don't think it's worth getting into for this PR.

Reply:
Please submit an issue.